api/base/filters.py

```diff
@@ -504,8 +505,10 @@ def preprints_queryset(self, base_queryset, auth_user, allow_contribs=True):
             admin_user_query = Q(node__contributor__user_id=auth_user.id, node__contributor__admin=True)
             reviews_user_query = Q(node__is_public=True, provider__in=get_objects_for_user(auth_user, 'view_submissions', PreprintProvider))
             if allow_contribs:
-                contrib_user_query = Q(is_published=True, node__contributor__user_id=auth_user.id, node__contributor__read=True)
-                query = default_query & (no_user_query | contrib_user_query | admin_user_query | reviews_user_query)
+                contrib_query_base = Q(node__contributor__user_id=auth_user.id, node__contributor__read=True)
+                not_reviews_contrib_query = contrib_query_base & Q(is_published=True)
+                reviews_contrib_query = contrib_query_base & (Q(provider__reviews_workflow__isnull=False) & ~Q(reviews_state=States.INITIAL.value))
+                query = default_query & (no_user_query | not_reviews_contrib_query | reviews_contrib_query | admin_user_query | reviews_user_query)
```

Comment:
This is a fun pile of logic that keeps on growing.

Reply:
Yeah, once there are multiple workflows this isn't going to work.

Reply:
Hmm, while waiting for travis it occurred to me that with #7767, all preprints for unreviewed providers are set to `accepted` when they're published, so I think we could do:
```python
contrib_user_query = ~Q(reviews_state=States.INITIAL.value) & Q(node__contributor__user_id=auth_user.id, node__contributor__read=True)
```
regardless of workflow. Does that seem reasonable?

Reply:
And it *might* reasonably hold for future workflows that a preprint should be visible to contributors once it's transitioned away from `initial`.

Reply:
Yeah, that would work. For some reason I was thinking `reviews_state` would be `null` for provider who don't use moderation but the default is `initial` and it's not a nullable field.